src/PPWCode.Host.Core.WebApi/TransactionalActionFilter.cs

```diff
@@ -173,8 +174,20 @@ protected override Task OnActionExecutingAsync(ActionExecutingContext context, C
             return Task.CompletedTask;
         }
 
-        private static bool IsSuccessStatusCode(int statusCode)
-            => (statusCode >= (int)HttpStatusCode.OK) && (statusCode <= 299);
+        private bool IsSuccessStatusCode(IActionResult actionResult)
```

Comment:
Instead of IActionResult, I prefer ActionExecutedContext context as parameter. In case of failure, you can report which action failed by name, also in which controller.